x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/authn/SuccessfulAuthenticationResponseMessageBuilder.java

```diff
@@ -45,32 +49,33 @@
  */
 public class SuccessfulAuthenticationResponseMessageBuilder {
 
-    private final SamlFactory samlFactory;
     private final Clock clock;
     private final SamlIdentityProvider idp;
+    private final XMLObjectBuilderFactory builderFactory;
 
-    public SuccessfulAuthenticationResponseMessageBuilder(SamlFactory samlFactory, Clock clock, SamlIdentityProvider idp) {
-        this.samlFactory = samlFactory;
+    public SuccessfulAuthenticationResponseMessageBuilder(Clock clock, SamlIdentityProvider idp) {
+        SamlUtils.initialize();
         this.clock = clock;
         this.idp = idp;
+        this.builderFactory = XMLObjectProviderRegistrySupport.getBuilderFactory();
```

Comment:
What's the reason for dropping the `SamlFactory` ?

Reply:
I thought that SamlUtils covers the functionality and purpose so there was no reason to have both and I moved relevant parts from SamlFactory to SamlUtils ( this is more or less what we do in the saml realm implementation ). 

If you feel that we should split somehow ( Instance with initialization for loggers in Factory and all static methods in Utils for instance ) I'm happy to discuss or just do it. I have no strong preference

Reply:
I found SamlUtils in security to have been a bit overloaded, and was hoping to avoid replicating it here. I'm not stuck on it, I just have an aversion to catch-all utility classes.

Reply:
Thinking this through again and seeing that I've made all methods in SamlUtils static already, I'm not sure what distinction would make sense. As in what should be in Utils and what should be in the Factory class. At 300 loc, SamlUtils doesn't seem too bloated atm.

I'm open to suggestions on what should be split away and back to a Factory class, or we could merge as is and spit these somehow once/if SamlUtils gets too bloated/overloaded

Reply:
This is definitely a personal preference thing, so I'm OK with merging as-is (though, in terms of preferences, I preferred the existing code).

Issues with Utility classes:
1. They become a catch-all dumping ground that becomes a cheap answer when something is needed, and avoids doing the harder design thinking. A good chunk of Utility class methods are working around broken class design that ideally would be fixed up front (but sometimes you have to because the broken class is in the JDK).
2. Related to the above, they often end up with a weird mix of responsibilities/behaviours. Because they're just chunks of code that _"do something, that someone thought was useful"_, they typically don't go through the process of _Is this the right behaviour for this method?_. 
3. Related to that, they can be hard to test well, unless they've got a clear interface & a single purpose (those are the best kind of utility method).
4. More importantly, it becomes hard to test code the _uses_ utility methods. They can't be mocked (because they're static), and it's often hard to trigger some of the behaviours that you need. As an example, while writing tests for SSL error messages, it was almost impossible to force PemUtils to throw certain exceptions so that I could verify they were handled neatly.
5. It can be hard to find the right method, because of the dumping ground problem. Ever tried to find the right helper method in amongst `XContentUtils`, `XContentParserUtils`, `XContentHelper`, `XContentTestUtils` ?

Now, you're probably thinking _Wait, `SamlUtils` and `SamlFactory` makes problem 5 **worse**_. And that's true, but `SamlInit` and `SamlFactory` were clear. It's when we decide to take the easy option and just have a `Utils` class that we lose clarity about what we each class is supposed to do.

However, as I said, personal preference. This definitely isn't a hill I'd die on. But those are the sorts of reasons why I tried to start with `SamlInit` and `SamlFactory` for this project.

Reply:
No need for anyone to die on this hill :) Given we've put in the time to climb it already though, I'll try and come up with something that you feel better with. I'll put the initialization logic in SamlInit to we can call `initialize()` on it and then move things back to a SamlFactory class which we can instantiate and call methods on.